### PR TITLE
TC master compatibility as of 9.2.0.43345

### PR DESCRIPTION
**Pull request by Smerdokryl**

### CODE REVIEW

File: LibTrinityCore-1.0/LibTrinityCore-1.0.lua [hunk 1]
```diff
@@ -39,15 +39,17 @@ end
 
 local frame = CreateFrame("Frame")
 frame:SetScript("OnEvent", function(self, event, prefix, message, channel, sender)
+  PlayerName = table.concat({UnitFullName("player")}, '-') -- "PlayerName-RealmName"
```
Comment (Treeston) [inline LibTrinityCore-1.0/LibTrinityCore-1.0.lua:42]: isn't this just `GetUnitName("player",true)`?

Comment (Treeston) [reply LibTrinityCore-1.0/LibTrinityCore-1.0.lua:42]: also you're polluting the global namespace, don't do that

Comment (Smerdokryl) [reply LibTrinityCore-1.0/LibTrinityCore-1.0.lua:43]: GetUnitName("player",true) just returns the unit name, without the -RealmName part.

Oopsie about the PlayerName thing - fixed

File: LibTrinityCore-1.0/LibTrinityCore-1.0.lua [hunk 2]
```diff
@@ -39,15 +40,17 @@ end
...
     self:RegisterEvent("CHAT_MSG_ADDON")
-    SendAddonMessage("TrinityCore", "p0000", "WHISPER", (UnitName("player")))
+    C_ChatInfo.RegisterAddonMessagePrefix("TrinityCore")
+    C_ChatInfo.SendAddonMessage("TrinityCore", "p0000", "WHISPER", PlayerName)
     return
   end
   assert(event == "CHAT_MSG_ADDON")
   if prefix ~= "TrinityCore" then return end
-  if sender ~= UnitName("player") then return end
+  if sender ~= PlayerName then return end
```
Comment (Shauren) [inline LibTrinityCore-1.0/LibTrinityCore-1.0.lua:53]: Does this always work? I dont know when client adds -Realm to name and when it doesn't

Comment (Smerdokryl) [reply LibTrinityCore-1.0/LibTrinityCore-1.0.lua]: I used `UnitFullName` which always returns the realm name, so yes.

However, both `UnitName` and `UnitFullName` might return the wrong name if the player is transformed by some spell, so instead we should use `UnitNameUnmodified`. Unfortunately, it does not guarantee a realm name like `UnitFullName` does, so I have also added `GetRealmName` to be 100% safe.

Comment (Treeston) [reply LibTrinityCore-1.0/LibTrinityCore-1.0.lua]: @Shauren's question was about whether `sender` is guaranteed to be the fullname form here.

Comment (Smerdokryl) [reply LibTrinityCore-1.0/LibTrinityCore-1.0.lua]: That seems to be the case in my experience, but I haven't been able to find any 'official' confirmation.

As of the latest commit, I could have it check against both `playerFullName` and `playerName`, though. Should I?

**Pull request merged**